src/data/set/basic.lean

```diff
@@ -185,9 +185,16 @@ by tauto
 
 /-! ### Lemmas about `mem` and `set_of` -/
 
-@[simp] theorem mem_set_of_eq {a : α} {p : α → Prop} : a ∈ {a | p a} = p a := rfl
+@[simp] theorem mem_set_of_eq {a : α} {p : α → Prop} : a ∈ {x | p x} = p a := rfl
```

Comment:
I assume the only reason we have this lemma in the first place is to avoid unnecessary `propext`?

Reply:
I believe it has something to do with `dsimp` which (IIRC) uses equalities proven by `rfl` but not any `iff` statements proven by reflexivity.

Reply:
I'm pretty sure `dsimp` knows about `iff.rfl`, but maybe it didn't use to.